### PR TITLE
Refactor number formatting

### LINKED ISSUE #53: Refactor number formatting
Number formatting got out of control over the years of dapp development. It only became more apparent as I worked on https://github.com/JoinColony/colonyDapp/pull/3922. Working on cdapp gives us a chance of fresh start and hopefully get it back in order. 

The initial plan is to create a `Numeral` component that would accept a raw BigNumberish value and format it according to [the proposal](https://www.notion.so/colony/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb).
If it becomes necessary, there should be a util method returning a formatted value. That comes with challenges of its own, including handling of tiny or huge numbers (we can't really return a nicely formatted engineering notation) which would need to be addressed.
The general rule of thumb for handling token values should be that only the raw values (expressed in the base unit/smallest denomination of a token) should be passed between components and in any case where the formatting is necessary, the `Numeral` component should be utilised.

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

This PR refactors number formatting to be provide a more streamlined process and follow the rules outlined in [the proposal](https://www.notion.so/colony/Number-formatting-proposal-2eb2b390861349938b9e30d9abb583fb)

In general, it tries to encapsulate all the formatting logic inside the `Numeral` component so that the formatting is more predicable and easier to reason about. It doesn't yet provide any way to imperatively format a number (eg. with an util function) - this is quite difficult to handle since, when using engineering notation, we need to return some additional HTML markup apart from just the value.

As discussed with @rdig, I will keep an eye out on any incoming PRs that require number formatting to make sure this solution is working for every case.

**Changes** 🏗

* Refactored `Numeral` component
* Updated all consumers to use new props
* Refactored `getFormattedTokenValue` util to limit its responsibility to shifting decimal point and avoid formatting (I'm open to suggestions on a better name for this function)

**Deletions** ⚰️

* Deleted number utils

Resolves #53

### CODE REVIEW

**Review by rdig — APPROVED** [review]
Yeah, looks very nice. :+1: 

I assume you ported most of the logic from the existing work you did in the Dapp.

I tested a couple of basic scenarios, all seem to work well. It goes into scientific notation going over 100 billion.

Screenshot from 2022-11-08 16-20-49
Screenshot from 2022-11-08 16-21-04
Screenshot from 2022-11-08 16-21-31
Screenshot from 2022-11-08 16-21-54
Screenshot from 2022-11-08 16-57-57

And I can render small values with up to a 5 decimal accuracy:

Screenshot from 2022-11-08 17-00-06

Anything under that shows as negative scientific notation

Screenshot from 2022-11-08 17-00-57

Overall pleased with how it turned out.

Please try to merge it as soon as possible, as I'd like to have this in prior to getting to the UI part of my #74 PR

File: src/utils/tokens.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import { BigNumber, BigNumberish } from 'ethers';
 import Decimal from 'decimal.js';
+import { BigNumber, BigNumberish } from 'ethers';
 
-import { minimalFormatter } from '~utils/numbers';
-import { TokenWithBalances } from '~data/index';
-import { DEFAULT_TOKEN_DECIMALS, SMALL_TOKEN_AMOUNT_FORMAT } from '~constants';
+// import { TokenWithBalances } from '~data/index';
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/utils/tokens.ts]: It's commented out in this function that I haven't edited:
image

I deleted the import and added a TODO for changing the `any` type

**Review by Julianahlk — APPROVED** [review]
Nice work Jakub! The new formatting is working very well 🙌
Screen Shot 2022-11-08 at 9 47 20 PM
Screen Shot 2022-11-08 at 9 48 03 PM

**Review by danbr — APPROVED** [review]
Looks good - Nice work.

**Review by chinins — APPROVED** [review]
Really nice - great work!

**Pull request merged**